crates/subspace-core-primitives/src/lib.rs

```diff
@@ -274,13 +272,27 @@ impl FlatPieces {
     pub fn into_inner(self) -> Vec<u8> {
         self.0
     }
+
+    /// Iterator over individual pieces as byte slices.
+    pub fn as_pieces(&self) -> impl ExactSizeIterator<Item = &[u8]> {
```

Comment:
There are a bunch of `pieces.as_pieces()` which just sounds like _pieces as pieces_, perhaps better to use `flat_pieces.as_pieces` in these cases. Just a thought, feel free to ignore it.

Reply:
There are a few places where it can be done, but I don't think it can be done everywhere while making perfect sense. Let's keep this in mind and maybe refactor in the future.